File: Models/Responses/Logs/ResponseLogAction.cs
using Models.Common.Enums;

namespace Models.Responses.Logs;

/// <summary>
/// 액션 로그 응답 클래스 
/// </summary>
public class ResponseLogAction
{
    /// <summary>
    /// 등록일
    /// </summary>
    public DateTime RegDate { get; init;}

    /// <summary>
    /// 등록자명 
    /// </summary>
    public string RegName { get; init;} = "";

    /// <summary>
    /// 내용 
    /// </summary>
    public string Contents { get; init;} = "";

    /// <summary>
    /// 카테고리 
    /// </summary>
    public string Category { get; init; } = "";
    
    /// <summary>
    /// 액션타입
    /// </summary>
    public EnumDatabaseLogActionType ActionType { get; set; }

}

File: BudgetBringer/src/models/responses/logs/response-log-action.ts
import {ResponseCommonWriter} from "../response-common-writer";
import {EnumDatabaseLogActionType} from "../../enums/enum-database-log-action-type";

/**
 * 액션 로그 응답 클래스
 */
export class ResponseLogAction extends ResponseCommonWriter {
  /**
   * 내용
   */
  contents: string;
  /**
   * 카테고리
   */
  category: string ;
  /**
   * 액션타입
   */
  actionType: EnumDatabaseLogActionType ;
}
